### DISCUSSION TITLE
How to properly backup running CDS?

### DISCUSSION THREAD

**Discussion by semenchukovps** [ovh/cds#6359] [Q&A]
Hi

Could you help me please, how to correctly back up my CDS instance?
Now i just export and import pipelines/workflows/encironment/etc. But in this case i lose webhooks, tokens and history.

Is it possible to backup database and restore it to new instance? As i figure out, there are some tokens between database and API.

**Comment by yesnault** [comment]
Hi. With a backup of your DB and configuration file (mandatory, to keep encryption keys), it will be ok to restore it elsewhere.

**Reply by semenchukovps** [reply]
@yesnault thx for you answer

but i have some issues with it

my commands for backup:
$ docker exec $(docker ps -aqf "name=user_cds-db_1") sh -c "pg_dumpall -U cds > /tmp/all_pg_dbs.sql"
$ docker cp $(docker ps -aqf "name=user_cds-db_1"):/tmp/all_pg_dbs.sql backup/
$ docker cp $(docker ps -aqf "name=user_cds-api_1"):/app/conf/conf.toml backup/

then i restore it:
$ docker cp backup/conf.toml $(docker ps -aqf "name=user_cds-api_1"):/app/conf/conf.toml
$ docker cp backup/all_pg_dbs.sql $(docker ps -aqf "name=user_cds-db_1"):/tmp/restore_pg_dbs.sql
$ docker exec $(docker ps -aqf "name=user_cds-db_1") sh -c "psql -U cds --dbname cds -f /tmp/restore_pg_dbs.sql"

and my pipelines, workflows, tokens seems to be restored (it's available in UI). but there is issue, as i figure out, with authentication. when i try to open workflow or tab with the list of ssh-keys, UI shows push-up notification "Internal server error". And the log message is:

`[ERROR] internal server error (caused by: unable to decrypt content (<nil><nil>proj-ssh-testjWq0qpzQPnUbl0lK0EI_bSlZ_aRnS09t7NN3O-i0cO7R8HZiK7fLPysY0ywHFe2SiwFnqp5w5qwIbtCP5B90aQ==): message authentication failed) action=api.(*API).getKeysInProjectHandler action_metadata_project_key=TEST auth_consumer_id=5b1d5266-0b95-4440-a2fb-7e57b4220d20 auth_session_iat=1668523400 auth_session_id=be54bec8-f88c-4de3-801b-84bf16aba9db auth_session_token= auth_user_id=763d5d86-78aa-4207-96c6-9d8d136bd4d8 auth_user_name=admin caller=github.com/ovh/cds/engine/service.WriteError deprecated=false handler=api.(*API).getKeysInProjectHandler ip_address=192.168.64.11:49120 method=GET request_id=34bb6e53-c619-4f00-bb80-246b142bfdfa request_uri=/project/TEST/keys route=/project/<project-key>/keys source_file=/tmp/790ad7b9a1a85167593986ed564384b7/run/engine/service/http.go source_line=157 stack_trace=message authentication failed`

Am i doing something wrong, or something else must be done?

Thank you
